google-cloud-core/src/main/java/com/google/cloud/ServiceOptions.java

```diff
@@ -73,7 +73,7 @@
   private static final String MANIFEST_ARTIFACT_ID_KEY = "artifactId";
   private static final String MANIFEST_VERSION_KEY = "Implementation-Version";
   private static final String ARTIFACT_ID = "google-cloud-core";
-  private static final String LIBRARY_NAME = "gcloud-java";
```

Comment:
Is Datastore discovery-based on Java? If not, we'll need to use the custom `x-goog-api-client` header. Since this is not on GAPIC yet, we'll just need to implement the entire header string for Datastore or we won't have any usage metrics for it.

Using UA is correct for the other three services.